src/rpc/rawtransaction.cpp

```diff
@@ -120,11 +122,18 @@ void TxToJSON(const CTransaction &tx, const uint256 hashBlock, UniValue &entry)
                 entry.pushKV("confirmations", 1 + chainActive.Height() - pindex->nHeight);
                 entry.pushKV("time", pindex->GetBlockTime());
                 entry.pushKV("blocktime", pindex->GetBlockTime());
+                confs = true;
             }
-            else
-                entry.pushKV("confirmations", 0);
         }
     }
+    // If the confirmations wasn't written with a valid block, then we have 0 confirmations.
+    if (!confs)
```

Comment:
why not put the following code in the `else` branch of the `if` above and avoid to use `conf` auxiliary variable?

Reply:
because there are 3 else locations where I'd have to put it...

Reply:
In `TxToJSON()` really? must have missed that. it's a nit by the way it does not matter.

Reply:
its a git diff thing... expand the missing lines and you'll see that the code is this:
```    if (!hashBlock.IsNull())
    {
...
        if (pindex)
        {
            if (chainActive.Contains(pindex))
            {
...
                confs = true;
            }
        }
    }
```